formats/dsd/http.go

```diff
@@ -0,0 +1,132 @@
+package dsd
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+var (
+	ErrMissingBody        = errors.New("dsd: missing http body")
+	ErrMissingContentType = errors.New("dsd: missing http content type")
+)
+
+const (
+	httpHeaderContentType = "Content-Type"
+)
+
+func LoadFromHTTPRequest(r *http.Request, t interface{}) (format uint8, err error) {
+	if r.Body == nil {
+		return 0, ErrMissingBody
+	}
+	defer r.Body.Close()
+
+	return loadFromHTTP(r.Body, r.Header.Get(httpHeaderContentType), t)
+}
+
+func LoadFromHTTPResponse(resp *http.Response, t interface{}) (format uint8, err error) {
+	if resp.Body == nil {
+		return 0, ErrMissingBody
+	}
+	defer resp.Body.Close()
```

Comment:
Again, according to the docs this check is pointless:

```
        // The http Client and Transport guarantee that Body is always
	// non-nil, even on responses without a body or responses with
	// a zero-length body. It is the caller's responsibility to
	// close Body. The default HTTP client's Transport may not
	// reuse HTTP/1.x "keep-alive" TCP connections if the Body is
	// not read to completion and closed.
```

Like for the `LoadFrom HTTPRequest` I'm not sure if it's really the responsibility of this function that the response body is closed. I'd rather consider this as a requirement for the user of this method. In either way, we should document the behavior!